v2.0.0-rc.1

Kevsy opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* documentation



#### What this PR does / why we need it:

This is the first release candidate PR for SED v2.0.0 . It only includes the updates necessary for a release PR:
- updates version numbers in YAML
- updates CHANGELOG.MD to reflect changes since last public release


#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #

#### Special notes for reviewers:



#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.



```
docs

```

hdamker commented:
@FabrizioMoggio Great to see the release PR, I will create the release review issue. It would be good if you:
* Touch also the checklist, versions in .feature files etc
* Add in CHANGELOG the PRs which have solved the issues (you can omit some of them btw if they are not relevant for API Consumers)

That will make the release review easier.

jgarciatovar commented:
LGTM

JoseMConde reviewed (dismissed):
LGTM

Kevsy commented:
> @FabrizioMoggio 

@hdamker I guess you mean me :grin:

Great to see the release PR, I will create the release review issue. It would be good if you:
> 
> * Touch also the checklist, versions in .feature files etc

Thanks! Changed API Readiness Checklist API version, added version to Feature name


> * Add in CHANGELOG the PRs which have solved the issues (you can omit some of them btw if they are not relevant for API Consumers)

I've added them as links after the relevant issues (those which affect API consumers)  but let me know if the formatting should be changed.

hdamker commented:
> Thanks! Changed API Readiness Checklist API version, added version to Feature name
> 
> > * Add in CHANGELOG the PRs which have solved the issues (you can omit some of them btw if they are not relevant for API Consumers)
> 
> I've added them as links after the relevant issues (those which affect API consumers) but let me know if the formatting should be changed.

@Kevsy yep, yes I meant you ... too many comments written already. 

Regarding the format: the important point is here that readers of the CHANGELOG have links to the PRs which they can easily open to see what actually was changed in the files to address the point (that also the reasons why one PR should address normally one issue/topic). There is no defined format, but most teams follow the format which is created by GitHub if you use "Generate release notes" e.g.

`<PR title> by @Kevsy in https://github.com/camaraproject/SimpleEdgeDiscovery/pull/78`

Normally you have to rewrite <PR title> by something from the "release note" part of the PR (if provided at all). Could be also that you would list the PR multiple times if it covered several issues.

Important point here is to have the full link to the PR, not only a relative one, so that the links work as well if someone downloaded the zip file or has cloned the repository locally.

Kevsy commented on code/API_definitions/simple-edge-discovery.yaml:
```diff
@@ -207,7 +207,7 @@ paths:
       security:
         - openId:
-            - simple-edge-discovery:edge-cloud-zones:read
+            - simple-edge-discovery:read
```
> Just for curiosity: why this (breaking) change? (I guess both options are correct)

@hdamker changed because the Design Guide says:

"
  * api-name is the API name specified as the base path, prior to the API version, in the servers[*].url property. For example, from /location-verification/v1, it would be location-verification.

  * resource is optional. **For APIs with several paths, it may include the resource in the path**. For example, from /qod/v1/sessions/{sessionId}, it would be sessions."
 
"
 
 Since SED has only one path,  the optional resource does not apply according to the wording above. As this release is breaking anyway (because of the move from GET to POST) it seems a good time to make the scope name compliant at the same time :)

Kevsy commented on CHANGELOG.md:
```diff
@@ -22,6 +23,48 @@ The below sections record the changes for each API version in each release as fo
+# r2.1 - rc
+
+This release candidate contains the definition and documentation of
+* [simple-edge-discovery v2.0.0-rc.1]()
```
```suggestion
* simple-edge-discovery v2.0.0-rc.1
```

Kevsy replied:
Removed accidental link markdown

rartych commented:
Typo detected in `.feature` file: “succesfully” → “successfully”  

> `And the device has been succesfully identified`

rartych commented:
@Kevsy Please commit the accidental link removal, then we can go with the release.

Kevsy commented on README.md:
```diff
@@ -23,11 +23,17 @@ Incubating API Repository to evolve and maintain the definitions and documentati
+* Note: Please be aware that the project will have frequent updates to the main branch. There are no compatibility guarantees associated with code in any branch, including main, until a new release is created. For example, changes may be reverted before a release is created. **For best results, use the latest public release**.
 
-Pre-releases of this sub project are available in https://github.com/camaraproject/SimpleEdgeDiscovery/releases
+* **The latest public release candidate is [r2.1 - rc](https://github.com/camaraproject/QualityOnDemand/releases/tag/r2.2), with the following API version:**
```
```suggestion
* **The latest public release candidate is [r2.1 - rc](https://github.com/camaraproject/SimpleEdgeDiscovery/releases/tag/r2.2), with the following API version:**
```

rartych reviewed (dismissed):
LGTM

Kevsy commented:
Thanks @rartych - 
@JoseMConde now ready for Codeowner review 👍

rartych reviewed (approved):
Re-approved

JoseMConde reviewed (approved):
LGTM. Re-approved.
